### PR TITLE
Allow external form state update

### PR DESCRIPTION
**Pull request by iondrimba**
This PR aims to make it possible to update any part of the component state from an external action/call.
It was partially possible but we only permitted updating the last step(user fields)

**CHANGELOG** :memo:

* Replaces implementation of  `form. updateUserFields` > `form.updateState` 
* Some linting and warnings removed

**PRINTS** :framed_picture:

### CODE REVIEW

File: spec/components/Factory.spec.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ const textArea = form.steps[REQUEST_FIELD].fields[5];
 const commonProps = {
   onFieldChange: () => {},
   onFieldBlur: () => {},
+  zipcodeUrlService: '',
```
Comment (iondrimba) [inline spec/components/Factory.spec.js]: Fix warning of required prop

Comment (rafaeldev) [reply spec/components/Factory.spec.js]: This field is duplicated with line 15? :thinking:

File: spec/components/Form/index.spec.js [hunk 2]
```diff
@@ -120,23 +121,27 @@ describe('Form', () => {
           values: [],
         }];
 
-        component.instance().updateUserFields(mock);
+        const last = data.steps.length - 1;
+        const userFields = data.steps[last];
```
Comment (rafaeldev) [inline spec/components/Form/index.spec.js]:
```suggestion
        const userFields = [...data.steps].pop();
```
_show-off_

**Comment by iondrimba** [comment]
@williamcosta @rafaeldev I removed the duplicated field `zipcodeUrlService` from the spec

File: src/index.dev.js [hunk 3]
```diff
@@ -3,17 +3,54 @@ import ReactDOM from 'react-dom';
...
+  post() {
+    this.form.formSubmit();
+  }
+
+  updateState(state) {
+    this.form.updateState(state);
+  }
+
+  init() {
+    ReactDOM.render(
```
Comment (eduardojmatos) [inline src/index.dev.js]: Could this render be on the constructor? I see that `jason.init()` is called on line 56, but nothing is executed before, unless that options are configured (line 41)

File: spec/components/Form/index.spec.js [hunk 4]
```diff
@@ -101,11 +101,12 @@ describe('Form', () => {
     });
   });
 
-  describe('.updateUserFields', () => {
+  describe('.updateState', () => {
```
[the remark this answers is not in the document]
Comment (eduardojmatos) [reply spec/components/Form/index.spec.js]: Nice!

File: src/index.dev.js (hunk 3 above)
Comment (iondrimba) [reply src/index.dev.js]: I think putting the render of the entire component inside the constructor makes it do too much work.
It removes the possibility to defer render. I like to keep constructors doing very little work, mostly only props assignments.

**Pull request merged**